src/providers/google/calendar-event.ts

```diff
@@ -0,0 +1,390 @@
+        eventHistory = eventHistory.concat(fetchedEvents);
+        totalEvents += fetchedEvents.length;
+
+        // Update the calendar's sync data with the latest rangeTracker state
+        calendarList[calendarIndex].syncData = rangeTracker.export();
+
+      }
+
+      syncPosition.thisRef = calendarList[(Math.min(calendarCount, this.config.calendarLimit) + calendarPosition) % calendarCount].sourceId; // Continue from the next calendar in the next sync
+      
+      // Finalize sync position and status based on event count
+      this.updateSyncPosition(
+        syncPosition,
+        totalEvents,
+        calendarCount
+      );
+
+      return {
+        results: calendarList.concat(eventHistory),
+        position: syncPosition,
+      };
+    } catch (err: any) {
+      console.error(err);
+      throw err;
+    }
+  }
+
+  private getCalendarPositionIndex(
+    calendarList: SchemaCalendar[],
+    syncPosition: SyncHandlerPosition
+  ): number {
+    const calendarPosition = calendarList.findIndex(
+      (calendar) => calendar.sourceId === syncPosition.thisRef
+    );
+
+    // If not found, return 0 to start from the beginning
+    return calendarPosition === -1 ? 0 : calendarPosition;
+  }
+
+  private async fetchAndTrackEvents(
+    calendar: SchemaCalendar,
+    rangeTracker: ItemsRangeTracker,
+    apiClient: calendar_v3.Calendar
+  ): Promise<SchemaEvent[]> {
+    // Validate calendar and calendar.id
+    if (!calendar || !calendar.sourceId) {
+      throw new Error('Invalid calendar or missing calendar sourceId');
+    }
+
+    // Initialize range from tracker
+    let currentRange = rangeTracker.nextRange();
+
+    let items: SchemaEvent[] = [];
+
+    while (true) {
+      // Fetch events for the current range using fetchEventRange
+      const events = await this.fetchEventRange(calendar, currentRange, apiClient);
+
+      if (!events.length) break;
+
+      // Add fetched events to the main list
+      items = items.concat(events);
+
+      // Break loop if events reached calendar limit
+      if (items.length > this.config.eventsPerCalendarLimit) {
+        // Mark the current range as complete and stop
+        rangeTracker.completedRange({
+          startId: events[0].start.dateTime,
+          endId: events[events.length - 1].end.dateTime
+        }, false);
+        break;
+      } else {
+        // Update rangeTracker and continue fetching
+        rangeTracker.completedRange({
+          startId: events[0].start.dateTime,
+          endId: events[events.length - 1].end.dateTime
+        }, false);
+
+        // Move to the next range
+        currentRange = rangeTracker.nextRange();
+      }
+    }
+
+    return items;
+  }
+
+  private updateSyncPosition(
+    syncPosition: SyncHandlerPosition,
+    totalEvents: number,
+    calendarCount: number,
+  ) {
+    if (totalEvents === 0) {
+      syncPosition.status = SyncHandlerStatus.ENABLED;
+      syncPosition.syncMessage = "No new events found.";
```

Comment:
`Stopping. No results found.` for consistency with Gmail.

Reply:
Fixed